site/metrics.md

```diff
@@ -0,0 +1,146 @@
+---
+title: Monitoring with Prometheus
+menu_order: 105 
+---
+
+Two endpoints are exposed: one for the Weave Net router, and, when deployed as
+a [Kubernetes Addon](/site/kube-addon.md), one for the [network policy
+controller](/site/kube-addon.md#npc).
+
+### Router Metrics
+
+The endpoint address is `localhost:6782`; the following metrics are
+exposed:
+
+* `weave_connections` - Number of peer-to-peer connections.
+* `weave_connection_terminations_total` - Number of peer-to-peer
+  connections terminated.
+* `weave_ips` - Number of IP addresses.
+* `weave_max_ips` - Size of IP address space used by allocator.
+* `weave_dns_entries` - Number of DNS entries.
+* `weave_flows` - Number of FastDP flows.
+
+### Kubernetes Network Policy Controller Metrics
+
+The endpoint address is `localhost:8686`; the following metric is
+exposed:
+
+* `weavenpc_blocked_connections_total` - Connection attempts blocked
+  by policy controller.
+
+# Static Configuration for Weave Net
+
+The following YAML fragment can be used in your Prometheus configuration to
+scrape the router metrics endpoint:
+
+```
+scrape_configs:
+- job_name: 'static-weave'
+  scrape_interval: 15s
+  static_configs:
+  - targets: ['localhost:6782']
+```
+
+# Discovery Configuration for Kubernetes
+
+If you're running Weave in conjunction with Kubernetes, it is possible
+to discover the endpoints for all nodes automatically. The following
+Kubernetes config will install and configure Prometheus 1.3 on your
+cluster and configure it to discover and scrape the Weave endpoints:
+
+```
+
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: prometheus
+data:
+  prometheus.yml: |-
+    global:
+      scrape_interval: 15s
+    scrape_configs:
+    - job_name: 'kubernetes-weave'
+      kubernetes_sd_configs:
+      - api_server:
+        role: pod
+      relabel_configs:
+      - source_labels: [__meta_kubernetes_namespace,__meta_kubernetes_pod_label_name]
+        action: keep
+        regex: ^kube-system;weave-net$
+      - source_labels: [__meta_kubernetes_pod_container_name,__address__]
+        action: replace
+        target_label: __address__
+        regex: ^weave;(.+)$
```

Comment:
Could you replace by `regex: ^weave;(.+?)(?::\d+)?$`, because exposing any `containerPort` in the weave-kube manifest would include that port in the `__address__` label?